corecito.py

```diff
@@ -50,20 +64,45 @@ async def main():
         logger.info(f'> Exploded {increase_percentage:.2f}%\nConsider updating CoreNumber to {deviated_core_number:.6f}')
 
       elif coreNumberIncreased(account.core_number, deviated_core_number, account.min_core_number_increase_percentage, account.max_core_number_increase_percentage):
-        logger.info(f'Increased {increase_percentage:.2f}% - excess of {excess:.6f} {account.core_number_currency} denominated in {account.base_currency}')
-        tx_result = round(excess * buy_price, account.max_decimals_buy)
-        logger.info(f'\n\n>>> Selling: {tx_result:.6f} {account.base_currency} at {buy_price} to park an excess of {excess:.6f} {account.core_number_currency}\n')
+
+        #Check if 'fiat' is True to adjust messages format and tx_result var and 'excess' has to be divided by the buy_price
+        if fiat:
+          logger.info(f'Increased {increase_percentage:.2f}% - excess of €{excess:.2f} denominated in {account.base_currency}')
+          tx_result = round(excess / buy_price, account.max_decimals_buy)
+          logger.info(f'\n\n>>> Selling: {tx_result:.6f} {account.base_currency} at €{buy_price} to park an excess of €{excess:.2f}\n')
+        else:
+          logger.info(f'Increased {increase_percentage:.2f}% - excess of {excess:.6f} {account.core_number_currency} denominated in {account.base_currency}')
+          tx_result = round(excess * buy_price, account.max_decimals_buy)
+          logger.info(f'\n\n>>> Selling: {tx_result:.6f} {account.base_currency} at {buy_price} to park an excess of {excess:.6f} {account.core_number_currency}\n')
+
         # Sell excess of base currency ie. => in ETH_BTC pair, sell excess BTC => Buy ETH
+        # If fiat, we sell the value previously calculated and stored on tx_result
         if (not config['safe_mode_on']):
-          await account.order_market_buy(tx_result, excess)
+          if fiat:
+            await account.order_market_sell(tx_result)
```

Comment:
If we make a sell order for fiat, shouldn't we also use the sell_price to calculate the excess, instead of the buy_price?

Reply:
The `buy_price` is the closest price to the last known price a bider is willing to pay. If we use Limit Orders, then we can use the `sell_price`, but as we are using Market Orders, the `buy_price` should be used.

Umm, now that you mention this, actually we should use a new `last_price` variable to calculate it? Due to the Market Orders nature.

Reply:
I think the last_price means the last transaction price, but that doesn't mean there is still an order open in the orderbook with that price anymore.

The buy and sell prices are the top of the orderbook from both sides. So a market order should always start by buying/selling the top of the order book and if the market order is bigger than what's available in the top order, then the next order will be consumed.

Reply:
Regarding the buy/sell price I describe the scenario with both pairs:
In ETH/BTC, coreNumberIncreased means the value of my BTCs in ETHs is higher than my core number so I have to buy some ETHs with my BTCs to reduce my amount of BTCs.
To buy some ETHs we have to look at the **sell_price**.

In BTC/EUR, coreNumberIncreased means the value of my BTCs in EURs is higher than my core number so I have to sell some BTCs to take down the EURs value of my BTC.
To sell some BTCs we have to look at the **buy_price**.

Can you please check if this is correct?
If this is the case, then the existing logic was incorrect and the buy/sell prices should be inverted for the non-fiat transactions.

Reply:
Exactly, if you execute `account.order_market_buy` you should take `sell_price` for calculations, as you buy from the sellers (ask)ing for the minimum price they are willing to accept.
So, in `coreNumberIncreased` it should be:
```
        if fiat:
          tx_result = round(excess / buy_price, account.max_decimals_buy)
        else:
          tx_result = round(excess * sell_price, account.max_decimals_buy)
```
because we execute:
```
        if (not config['safe_mode_on']):
          if fiat:
            await account.order_market_sell(tx_result) <---- as we sell, we calculate with buy_price for fiat
          else:
            await account.order_market_buy(tx_result, excess) <--- as we buy, we calculate with sell_price for crypto
```

Reply:
Perfect, double better your PR, it helped find bugs in the existing script and it was still making profits 😅 😅 😅 
Would you mind the change/fix in this PR?